src/main/java/uk/org/tombolo/field/value/TimeseriesMeanValueField.java

```diff
@@ -0,0 +1,89 @@
+
+package uk.org.tombolo.field.value;
+
+import org.json.simple.JSONObject;
+import uk.org.tombolo.core.Subject;
+import uk.org.tombolo.core.TimedValue;
+import uk.org.tombolo.core.utils.TimedValueUtils;
+import uk.org.tombolo.field.IncomputableFieldException;
+import uk.org.tombolo.field.SingleValueField;
+import uk.org.tombolo.recipe.AttributeMatcher;
+
+import java.util.List;
+
+
+public class TimeseriesMeanValueField extends BasicValueField implements SingleValueField {
+    public TimeseriesMeanValueField(String label, AttributeMatcher attribute) {
+        super(label, attribute);
+    }
+
+    @Override
+    public String valueForSubject(Subject subject, Boolean timeStamp) throws IncomputableFieldException {
+        return Double.toString(getTimedValue(subject).getMeanValue());
+    }
+
+    @Override
+    public JSONObject jsonValueForSubject(Subject subject, Boolean timeStamp) throws IncomputableFieldException {
+        MeanValueObject meanValue = getTimedValue(subject);
+        JSONObject obj = new JSONObject();
+        if (null != timeStamp && !timeStamp) {
+            obj.put(null != this.label ? this.label : "value",
+                                            meanValue.getMeanValue());
+            return obj;
+        }
+        obj.put("timestamp", meanValue.getDateTimeRange());
+        obj.put("value", meanValue.getMeanValue());
+        return withinMetadata(obj);
+    }
+
+    private MeanValueObject getTimedValue(Subject subject) throws IncomputableFieldException {
+        List<TimedValue> timedValue = TimedValueUtils.getBySubjectAndAttribute(subject, getAttribute());
+        double addValue = 0.0;
```

Comment:
If you like, it could be something like:
```java
if (!timedValue.isEmpty()) {
        double meanValue = timedValue.stream().mapToDouble(i -> i.getValue()).average().getAsDouble();
        String timeStamp = timedValue.get(0).getId().getTimestamp() + " - " + 
            timedValue.get(timedValue.size() - 1).getId().getTimestamp();
} else {
     throw new IncomputableFieldException("Cannot compute because etc etc");
}
```

Reply:
done